### DISCUSSION TITLE
AC BO Hackathon manuscript brainstorming

### DISCUSSION THREAD

**Discussion by sgbaird** [AC-BO-Hackathon/ac-bo-hackathon.github.io#157] [General]
## Possible Titles
1. Bayesian Optimization Hackathon for Chemistry and Materials: Outcomes, Methods, and Future Directions

**Table**: list of projects and project types, with links to corresponding website project page, repos, videos, and social media posts. Perhaps also a link to "post a comment" (i.e., open a GitHub issue), and a link to a Zenodo snapshot of their repository. The description can have a link to a Zenodo snapshot of the hackathon website.

**Figure**: geographical map of the participation, maybe color coded by sector (Google maps or similar) It would be great to have an interactive counterpart to a publication figure (will need to track down that x sheets Google docs extension). This could have both organization and participant info. EDIT: @mehradans92 planning to work on this, likely with NetworkX, though maybe with https://thexs.ca.

**Figure**: cross-connections within teams geographically, maybe color coded by sector. EDIT: @mehradans92 planning to work on this, likely with NetworkX

**Figure**: bar charts
- distribution of sectors (academia vs. industry vs. government)
- distribution of country

**Table**: raw data of institutions that were represented

**Figure**: Gather Town event space (keynote, foyer, breakout, poster session, Easter egg).

**Text**: Anecdotes of personal connection that were formed (e.g., Andy Anker and Jakob Zeitler)

**Text**: Selected responses from follow-up survey related to what was learned or gained.

**Text**: "Methods" section of the hackathon. YouTube Livestream, GitHub classroom, Gather Town, GitHub website repo, slack. Team forming prior to hackathon (and the need to make LOTS of personal invites, including many follow-ups to make this happen in advance of the hackathon). Gavel/Sendgrid. Eventbrite. Feature in the website template that auto-switches web content for pre- vs. post-hackathon.

**Table**: table of project winners with corresponding mean and sigma squared values from project judging. In caption, number of judges and number of votes cast. Mention how judges were chosen.

**Text**: Choice of judges in terms of prior experiences and expertise affects the judging process significantly.

**Text**: Review Kevin's paper for additional ideas, and make sure to cite his paper. Mention the heavy templating and adaptation of the hackathon. Note that high-level structure was by topic, and then having individual projects within each topic. Also, great quote: "Therefore, one should expect an LLM to be as good as a rubber duck—if not drastically more effective"

**Text**: Best outcomes and lessons learned for next time. The poster session went really well in the end. Would have been better to number in advance. Allowing for free-form content between participants is important (i.e., the whiteboards, both for project formation and for poster sessions). Reduces burden on organizers, for example. Confusion about poster judging vs. final submission. Attrition from the Eventbrite, possibly related to timezones, except the choice of time may have helped in terms of geographical diversity. Separate instructions/invitations from people just wanting to watch the keynotes and those who want to fully participate might have been good. Would have been better to have a GitHub actions check from the start. It was set up to be scaleable. Gather town also helped level the playing field for geographically diverse teams and and colocated teams. Typically, one member from a colocated team could stay logged in. Making backups of the gather town event space would have been wise.

**Text**: Mention tips and tricks for running hackathons. Mention use of GitHub Classroom, pull requests for project files, etc. Link to the GitHub Classroom organization.

**Text**: Mention the benchmarks that were made available on HuggingFace, and how this can help standardize things. Mention desire to make these kaggle competitions. Maybe cite the recent Leash Bio competition.

**Figure**: benchmarks (picture of the interface of one of the benchmarks)

**Table**: Benchmark title, domain, complexity notes, link to benchmark

**Text**: source code snippet for benchmark API. Cite matsci-opt-benchmarks and CrabNet hyperparameter paper.

**Text**: Importance of keeping a broad scope for the hackathon (per recommendations in hackathon literature).

**Text**: Links to the keynote videos and the livestream. Use of YouTube live streaming.

**Text**: (Intro) why is Bayesian optimization for chemistry and materials important.

## CRediT form

- Ask for acknowledgments, conflicts of interest, etc.
- ORCiD IDs
- Affiliations
- Emails

## References

- Kevin Jablonka & Ben Blaiszik, LLM manuscript
- Some workshop-style manuscripts for SDLs/automation
- [A framework for materials informatics education through workshops](https://doi.org/10.1557/s43577-023-00531-6)
- Prior hackathon literature?
- [An Instrument Assembly and Data Science Lab for Early Undergraduate Education](https://doi.org/10.1021/acs.jchemed.2c01072) (JCE)
- Interactive Python Notebook Modules for Chemoinformatics in Medicinal Chemistry (JCE)

Text: what is the overlap between the participants in the LLM hackathon and the participants in the BO Hackathon? Use ChatGPT to find the overlap (just copy-paste from author list https://pubs.rsc.org/en/content/articlelanding/2023/dd/d3dd00113j).

## Considerations

Should we allow figures from individual projects, or should we just direct people to the website pages? I could make a Zenodo snapshot of the website repository, in which case directing to the hackathon website still has some permanence.

It may be worth asking people to publish Zenodo releases (easy through AC-BO-Hackathon repo) and link to those in the main table. https://zenodo.org/account/settings/github/

## Potential suggested reviewers

- Tonio?
- Jae?
- Hackathon queen?
- Tejs
- Jens
- Anjuli
- Mark

**Task**: Look to authors of related articles.

## Possible Journals

- Digital discovery
- Chemistry of Materials
- CellPress Device (Marshall Brennan)
- CellPress Matter
- Journal of Chemical Education
- Journal of Open Source Education
- Nature materials (perspective)

**Comment by mehradans92** [comment]
Hey @sgbaird! Not sure why https://thexs.ca/ does not have export options. Instead, I made this prototype with `CartoPy`.  What do you think?
![download](https://github.com/AC-BO-Hackathon/ac-bo-hackathon.github.io/assets/51170839/bd3f6ac5-4dd6-476e-8dce-57022f832983)
Or maybe a little fancier like:
![download](https://github.com/AC-BO-Hackathon/ac-bo-hackathon.github.io/assets/51170839/921e0ae5-738d-40ca-a47e-a5bca107cbf7)


I think it can also have an interactive counterpart for this one too. Will add color code by sector but need to make the dataset for it probably manually, unless we have it already. Similar idea for cross-connections within teams geographically that will be colored directed graphs.

**Reply by sgbaird** [reply]
Awesome! I think either style would be great. I think an interactive one would be really good. Maybe we can post about getting help with grabbing the data from the project files and and getting institutional addresses. Do you mind posting there?

**Reply by mehradans92** [reply]
Yeah sure. I'll figure out the right format and then ask everyone to fill in a Google spreadsheet :)

**Comment by mehradans92** [comment]
So the Figure: bar charts can be also easily made with the same dataset that is going to be used for the first two figures. I think most of the suggestions are great. Something that might be interesting could be categorizing the projects based on how BO was used; hyperparameter tuning and efficient parameter search, better exploration of the chemical design space, etc.

**Comment by sgbaird** [comment]
Top priorities for completion:

1. Project table (via csvreader automation)
2. authors.tex (via csvreader automation)
3. 3-4 sentence intro from each project page
4. 3-4 sentence outcome descriptions for each project (via GPT-4o, @mehradans92)
5. Update world map
6. CRediT statement
7. Conclusion / touch up text and flow
